### PR TITLE
Parse list value for `terminado_settings`

### PR DESCRIPTION
**Pull request by krassowski**
Fixes https://github.com/jupyterlab/jupyterlab/issues/12540. Allows configuring `terminado_settings` from command line with:

```
--ServerApp.terminado_settings="shell_command=['/bin/zsh']"
```

or if users are willing to accept the default tokenization by `shlex.split` with:

```
--ServerApp.terminado_settings="shell_command='/bin/zsh'"
```

Are there tests for configuration settings?

For context it is used in: [`jupyter_server_terminals`](https://github.com/jupyter-server/jupyter_server_terminals/blob/b8fea6404b8b5e5ee6ea8d69d3b680ee440e0508/jupyter_server_terminals/app.py#L44)

### CODE REVIEW

**Comment by blink1073** [comment]
This seems like a reasonable change for now.  Ideally we'd have a custom parser for each of the allowed fields.  Here's an example of a config setting [test](https://github.com/jupyter-server/jupyter_server/blob/323def5b36f2906048d8c2c69771a5ea6a118330/tests/test_serverapp.py#L286).

File: tests/test_terminal.py [hunk 1]
```diff
@@ -258,3 +258,22 @@ async def test_culling(jp_server_config, jp_fetch):
             await asyncio.sleep(1)
 
     assert culled
+
+
+@pytest.mark.parametrize(
+    "terminado_settings,expected_shell",
+    [
+        ("shell_command=['/path/to/shell', '-l']", ["/path/to/shell", "-l"]),
```
Comment (krassowski) [inline tests/test_terminal.py:266]: Using 1-element list here because once https://github.com/jupyter-server/jupyter_server_terminals/pull/42 gets merged and released I will add another test case of using a string in a follow-up.

Comment (blink1073) [reply tests/test_terminal.py:266]: Released!

Comment (krassowski) [reply tests/test_terminal.py]: Thanks, that was quick! Turns out parsing `Union`s from command line is broken in traitlets, I opened https://github.com/ipython/traitlets/pull/773 and tested that this test passes with traitlets patch.

Comment (krassowski) [reply tests/test_terminal.py]: Just linking to a follow up to traitlets patch which is fixing some edge cases: https://github.com/ipython/traitlets/pull/774 for convenience.

**Comment by blink1073** [comment]
https://github.com/ipython/traitlets/releases/tag/5.4.0

**Review by blink1073 — DISMISSED** [review]
Thanks!   Failures are unrelated

**Comment by krassowski** [comment]
Thanks! [Test Minimum Versions](https://github.com/jupyter-server/jupyter_server/actions/runs/3041460249/jobs/4898594414#logs) failure is relevant (other tests also fail on  https://github.com/jupyter-server/jupyter_server/pull/971). Should I:
- mark the test case for skipping on older traitlets, or
- bump the minimum version of traitlets to 5.4 (currently 5.1)?

**Comment by blink1073** [comment]
Oh, the minimum version check is related.  We'll need a version guard on the new test for the traitlets version.

**Comment by blink1073** [comment]
> mark the test case for skipping on older traitlets, or

This one  :smile:

**Comment by blink1073** [comment]
Thanks again!

**Comment by vidartf** [comment]
@krassowski Note that the test that was added was failing locally on Windows for me. This change to the test make it pass, but I'm not sure if this means the bug is fixed on Windows or not? https://github.com/jupyter-server/jupyter_server/pull/983#discussion_r1012845926

**Comment by westurner** [comment]
Is there already a way to use busybox ash (or bash) compiled to WASM with this setting?

emscripten-forge could build busybox, bash, and zsh, I think
